src/Dlp/V2beta1/Gapic/DlpServiceGapicClient.php

```diff
@@ -193,16 +194,31 @@ public static function resultName($result)
     /**
      * Parses a formatted name string and returns an associative array of the components in the name.
      * The following name formats are supported:
-     * - inspect/results/{result}.
+     * Template: Pattern
+     * - result: inspect/results/{result}.
+     *
+     * The optional $template argument can be supplied to specify a particular pattern, and must
+     * match one of the templates listed above. If no $template argument is provided, or if the
+     * $template argument does not match one of the templates listed, then parseName will check
+     * each of the supported templates, and return the first match.
      *
      * @param string $formattedName The formatted name string
+     * @param string $template      Optional name of template to match
      *
      * @return array An associative array from name component IDs to component values.
+     *
+     * @throws ValidationException If $formattedName could not be matched.
      * @experimental
      */
-    public static function parseName($formattedName)
+    public static function parseName($formattedName, $template = null)
     {
-        foreach (self::getPathTemplateList() as $pathTemplate) {
+        $templateMap = self::getPathTemplateMap();
+
+        if (isset($templateMap[$template])) {
```

Comment:
I think an exception here if `$templateMap[$template]` is not set is preferable. As it stands, one could continue under the assumption they've optimized to avoid the loop when in reality all they've done is provided an invalid value.